connector/client/client.go

```diff
@@ -84,18 +83,30 @@ func (client *ConnectorClient) Post(target url.URL, activity schema.Activity) er
 
 	replyClient := &http.Client{}
 
-	resp, err := replyClient.Do(req)
-	if err != nil || resp.StatusCode != http.StatusCreated {
+	return client.checkRespError(replyClient.Do(req))
+}
+
+func (client *ConnectorClient) checkRespError(resp *http.Response, err error) error {
+	allowedResp := []int{http.StatusOK, http.StatusCreated, http.StatusAccepted}
+	if err != nil {
 		return customerror.HTTPError{
-			StatusCode: resp.StatusCode,
-			HtErr:      err,
-			Body:       resp.Body,
+			HtErr: err,
 		}
 	}
-
 	defer resp.Body.Close()
 
-	return nil
+	// Check if resp allowed
+	for _, code := range allowedResp {
+		if code == resp.StatusCode {
+			return nil
+		}
+	}
+
+	return customerror.HTTPError{
+		HtErr:      errors.New("invalid response"),
+		StatusCode: resp.StatusCode,
+		Body:       resp.Body,
```

Comment:
Body if too large we might end up embedding too much content in HTTPError.

Reply:
Make sense. Let's skip embedding response body